sdk/go/common/workspace/templates_ai.go

```diff
@@ -0,0 +1,148 @@
+// Copyright 2016-2018, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package workspace
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"os"
+	"strings"
+)
+
+// Represents the body of the request to the Pulumi AI service to retrieve a project template.
+type AIProjectRequestBody struct {
+	ConversationID string `json:"conversationId"`
+	Model          string `json:"model"`
+}
+
+// Represents the response from the Pulumi AI service to retrieve a project template.
+type AIProjectPackageResponse struct {
+	Manifest []string        `json:"manifest"`
+	Files    []AIFilePayload `json:"files"`
+}
+
+// The shape of an individual file in the response from the Pulumi AI service to retrieve a project template.
+type AIFilePayload struct {
+	Name    string `json:"name"`
+	Content string `json:"content"`
+}
+
+func isAITemplateURL(templateNamePathOrURL string) bool {
+	return strings.HasPrefix(templateNamePathOrURL, "ai")
+}
+
+func retrieveAITemplates(templateURL string) (TemplateRepository, error) {
+	var err error
+	// Create a temp dir.
+	var temp string
+	if temp, err = os.MkdirTemp("", "pulumi-template-"); err != nil {
+		return TemplateRepository{}, err
+	}
+
+	parsedURL, err := url.Parse(templateURL)
+	if err != nil {
+		return TemplateRepository{}, err
+	}
+
+	var fullPath string
+	if fullPath, err = RetrieveAITemplateFolder(parsedURL, temp); err != nil {
+		return TemplateRepository{}, fmt.Errorf("failed to retrieve ai folder: %w", err)
+	}
+
+	return TemplateRepository{
+		Root:         temp,
+		SubDirectory: fullPath,
+		ShouldDelete: true,
+	}, nil
+}
+
+func RetrieveAITemplateFolder(templateURL *url.URL, tempDir string) (string, error) {
+	var projectPackageResponse AIProjectPackageResponse
+	serviceURL, conversationID := mapAITemplateURLToServiceURL(templateURL)
+	fetchNoticeLength, _ := fmt.Printf("Fetching project template for conversation %s from Pulumi AI...", conversationID)
```

Comment:
`logging.Info`, please don't assume its clean to write direct to stdout from where this is called (e.g. if we add this to automation api)

Reply:
What is the default verbosity setting for Pulumi re: logging? Setting it to V(1) results in no logs being written at all by default, and the UX is not nearly as good if we print nothing at all while doing this.

Reply:
In fact, nothing at all is printed even when `-v 10` is passed.

Reply:
Nothing is printed even when `-v 10` is passed while calling `logging.V(1).Infof`

Reply:
Default verbosity is 0, but by default logs don't write to stdout they write to a file. If you want to write to the user here we need to thread a diag.Sink into these calls. I would strongly suggest doing that change afterwards and also seeing where it makes sense to use that diag.Sink in the git and http template fetchers as well so they're all consistent.

Reply:
Omitting any logging here for now.